---
title: "Plotting C victoriae RiPP regions with transposons"
author: "Megan McDonald"
date: "02/08/2019"
output:
  pdf_document: default
  word_document: default
  html_document: default
editor_options:
  chunk_output_type: console
---

```{r setup}
knitr::opts_chunk$set(echo = TRUE)
```


# Run Lastz and convert text output into DNAcomp files for plotting in genoPlotR libary
##Exampe lastz run:

lastz Cvic_tig07.fasta Cvic_tig16.fasta --entropy \
--format=general:name1,strand1,start1,end1,length1,name2,strand2,start2+,end2+,length2,score,identity \
--markend --gfextend --nochain --gapped --step=1 --strand=both --output=Tig07_Tig16_out_nogap.txt \
--identity=90 --continuity=80 --matchcount=1000


```{r}

library(genoPlotR)
library(tidyverse)

raw_ident_comp<-read.delim("Cvictig07vstig16_nochain80_2kb.txt")
write_delim(raw_ident_comp,"Cvic07_Cvic16ident_80_2kb.comp",delim = "\t")

Chr_comp<-read_comparison_from_tab("Cvic07_Cvic16ident_80_2kb.comp", header=T)
Chr_comp$direction<-Chr_comp$strand


```
## Need to make file for important gene locations, i.e. RiPP genes, Copper Amine Oxidase
## Exported gff file from Geneious and edited manually in excel to make genes and transposon .seg files

## Make DNA alignment (comparisons) into a list
```{r}

Componly<-list(Chr_comp)

```

## Set X lims for whole chromosomes
```{r}
start.second=1
end.second=316216
start.main=1600000
end.main=2111100
xlims1 <- list(c(start.main, end.main), c(start.second, end.second))

```

## Set X lims for figure 2C
```{r}
start.second=160000
end.second=316216
start.main=2000000
end.main=2111100
xlims2 <- list(c(start.main, end.main), c(start.second, end.second))

```


## Plot everything with TEs
```{r}
CV07.annot.TE<-data.frame(read.delim("Tig07_gene_TE.segs", sep="\t", header=F))
names(CV07.annot.TE)<-c("name","start","end","strand","col","fill","gene_type")
CV07.gene_TE.seg<-dna_seg(CV07.annot.TE)

CV16.annot.TE<-data.frame(read.delim("Tig16_gene_TE.segs", sep = "\t", header = F))
names(CV16.annot.TE)<-c("name","start","end","strand","col", "fill","gene_type")
CV16.gene_TE.seg<-dna_seg(CV16.annot.TE)

GenesandTEs<-list(CV07.gene_TE.seg,CV16.gene_TE.seg)
names(GenesandTEs)<- c("Cvic Tig07","Cvic Tig16")
Componly<-list(Chr_comp)


plot_gene_map(dna_segs = GenesandTEs , comparisons = Componly, gene_type = "side_blocks", 
              global_color_scheme = c("identity", "auto","red_blue",0.8),
              dna_seg_scale = T, xlims=xlims2, scale=F, 
              main = "A",main_pos = "left",plot_new = T, legend = T)

```

```{r}
cairo_pdf(paste0("FigureXX.pdf"), width = 5, height = 2)
pushViewport(viewport(layout=grid.layout(1,1, heights=unit(c(1,1), rep("null",1))), name="Synteny between Tig07 and Tig16"))
## Panel A
pushViewport(viewport(layout.pos.row = 1, name="panelC"))
plot_gene_map(dna_segs = GenesandTEs , 
              comparisons = Componly, 
              gene_type = "side_blocks", dna_seg_scale= T, 
              scale=F, xlims=xlims2, 
              global_color_scheme = c("identity","auto","blue_red", 0.8), main="C", main_pos="left", legend=T)
dev.off()

```



